cpp/arcticdb/storage/mongo/mongo_client_wrapper.hpp

```diff
@@ -13,21 +13,9 @@
 
 namespace arcticdb::storage::mongo {
 
-template<class Output, class Result>
-struct MongoResult{
-    std::variant<Output, Result> result;
+struct UpdateResult { std::optional<int> modified_count; };
```

Comment:
A small doc on what `std::nullopt` means would be nice? E.g. "std::nullopt signals that the update failed (mongo docs are unclear on what type of error)".

Similar thing for DeleteResult.

Also happy to leave as is if this is going to get clearer in a follow up PR by using something shared like a `StorageResult`